### DISCUSSION TITLE
Script to update all installed servers

### DISCUSSION THREAD

**Discussion by tinny-blueleo** [williamboman/nvim-lsp-installer#749] [Q&A]
Hi friends!  I have a Packer script to keep my plugins up to date and wanted to bundle in a call to lua to make sure all of my installed LSP servers were updated (similar to picking 'U' in the LSP installer menus).  Is there a function or command I can call to accomplish this the right way?  Thank you!

**Comment by bryant-the-coder** [comment]
can u share ur packer script?

**Comment by tinny-blueleo** [comment]
Sure it's:

`nvim --headless -c 'autocmd User PackerComplete quitall' -c 'PackerSync' -c 'TSUpdate'`

**Comment by williamboman** [comment]
Hello! With https://github.com/williamboman/mason.nvim (which will succeed nvim-lsp-installer soon) this is pretty straight forward:

```lua
local mason_registry = require("mason-registry")

for _, pkg in ipairs(mason_registry.get_installed_packages()) do
    pkg:check_new_version(function (success, new_version)
        if success then
           -- Optionally do some logic based on `new_version`
           -- (which is a table { name: string, current_version: string, latest_version: string })
           pkg:install()
        end
    end)
end
```

**Reply by tinny-blueleo** [reply]
Thanks so much William, this is awesome.  What are the main differences between LSP installer and Mason?  It looks like Mason will install more things than just LSP servers.

**Reply by AlixShahid** [reply]
Mason is superset of nvim-lsp-installer meaning Mason is not just limited to managing LSPs, it can manage  ```Debugger Adapter Protocol(DAP)s```  ```Linters```  and ```Formatters``` too.
